hw-4/assets/index.js

```diff
@@ -0,0 +1,138 @@
+(async function() {
+    const $list = document.getElementById('list');
+    const $add = document.getElementById('add');
+    const $cancel = document.getElementById('cancel');
+    const $form = document.getElementById('form');
+    const $addBlock = document.getElementById('add-block');
+    let messages = [];
+
+    $list.addEventListener('click', (e) => {
+        if (e.target.classList.contains('edit')) {
+            setEditMode(e.target);
+        }
+        else if (e.target.classList.contains('cancel')) {
+            removeEditMode(e.target);
+        }
+        else if (e.target.classList.contains('delete')) {
+            deleteMessage(e.target);
+        }
+        else if (e.target.classList.contains('submit')) {
+            editMessage(e.target);
+        }
+    });
+
+    $cancel.addEventListener('click', (e) => {
+        $addBlock.classList.remove('edit');
+    });
+
+    $add.addEventListener('click', (e) => {
+        $addBlock.classList.add('edit');
+    });
+
+    $form.addEventListener('submit', async (e) => {
+        e.preventDefault();
+
+        const { author: { value: author }, text: { value: text } } = e.target.elements;
+
+        await sendRequest({
+            url: 'http://localhost:8000/messages',
```

Comment:
Поскольку файл отдаеться там же сервером, что и обрабатывает API, то можно так:
```suggestion
            url: '/messages',
```